File: app/src/main/java/com/konopko/stocktest/ViewModelAskTickers.java
package com.konopko.stocktest;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.google.gson.Gson;

import java.util.List;

import timber.log.Timber;

public class ViewModelAskTickers extends ViewModel {

    private MutableLiveData<Boolean> dataNext;

    public LiveData<Boolean> getDataNext(){
        if (dataNext == null)
            dataNext = new MutableLiveData<>();
        return dataNext;
    }

    public void next(@NonNull List<String> tickers){
        Timber.d("clickNext tags = %s", new Gson().toJson(tickers));
        if (!tickers.isEmpty()) {
            searchTickers(tickers);
            dataNext.setValue(true);
        }
    }

    private void searchTickers(@NonNull List<String> tickers){
        RepositoryTicker.getInstance().findTickers(tickers);
    }

}

File: app/src/main/java/com/konopko/stocktest/ViewModelAskTickers.kt
package com.konopko.stocktest

import androidx.lifecycle.SavedStateHandle
import androidx.lifecycle.ViewModel
import com.google.gson.Gson
import com.konopko.stocktest.ext.LiveEvent
import com.konopko.stocktest.repository.IRepositoryTicker
import org.koin.core.KoinComponent
import org.koin.core.inject
import timber.log.Timber

class ViewModelAskTickers constructor() : ViewModel(), KoinComponent {

    private val repositoryTicker : IRepositoryTicker by inject()

    private lateinit var state : SavedStateHandle //todo use this to save the view state

    val dataNext = LiveEvent<Boolean>()

    constructor(savedStateHandle: SavedStateHandle): this(){
        this.state = savedStateHandle
    }

    fun next(tickers : List<String>){
        Timber.d("clickNext tags = %s", Gson().toJson(tickers))
        if (!tickers.isEmpty()) {
            searchTickers(tickers)
            dataNext.setValue(true)
        }
    }

    private fun searchTickers(tickers: List<String>) {
        repositoryTicker.findTickers(tickers)
    }

}